app/src/main/java/com/celzero/bravedns/receiver/DeviceLockService.kt

```diff
@@ -66,9 +66,9 @@ class DeviceLockService  : Service(){
             this.stopSelf()
         } else {
             if (isProtected && isLocked) {
-                if (persistentState.getFirewallModeForScreenState() && !persistentState.getScreenLockData()) {
+                if (persistentState.screenState && !persistentState.isScreenOff) {
                     if(DEBUG) Log.d(LOG_TAG,"DeviceLockService : Screen lock detected at $delayIndex")
-                    persistentState.setScreenLockData(true)
+                    persistentState.isScreenOff = true
```

Comment:
If the `isScreenOff` is not persisted then the value is getting lost when the VPN is restarted. 
We were having it in-memory earlier and then changed to shared pref because of the issue.

Reply:
Created issue to remove isScreenOff from PersistentState. https://github.com/celzero/rethink-app/issues/304